spec/models/user_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validation test', :aggregate_failures do
+    it 'username presence' do
+      user = User.new(password: 'password', password_confirmation: 'password').save
+      expect(user).to eq(false)
+    end
+
+    it 'password presence' do
+      user = User.new(username: 'nimblehq', password_confirmation: 'password').save
+      expect(user).to eq(false)
+    end
+
+    it 'password confirmation presence' do
```

Comment:
For this kind of validation, we can use https://github.com/thoughtbot/shoulda-matchers @gutakk 😃

Reply:
Fixed in https://github.com/gutakk/ruby-google-scraper/pull/4/commits/137346c50a11344e0e153aab0d4daa4bcacce1f7